server/src/main/kotlin/org/jetbrains/bsp/bazel/server/sync/DependencyMapper.kt

```diff
@@ -0,0 +1,52 @@
+package org.jetbrains.bsp.bazel.server.sync
+
+import ch.epfl.scala.bsp4j.MavenDependencyModule
+import ch.epfl.scala.bsp4j.MavenDependencyModuleArtifact
+import org.jetbrains.bsp.bazel.server.sync.model.Library
+import org.jetbrains.bsp.bazel.server.sync.model.Module
+import org.jetbrains.bsp.bazel.server.sync.model.Project
+import java.util.*
+import kotlin.collections.HashSet
+
+object DependencyMapper {
+
+    fun extractMavenDependencyInfo(lib: Library): MavenDependencyModule? {
+        if (lib.outputs.isEmpty()) return null
+        val jars = lib.outputs.map { uri -> uri.toString() }.map {
+            MavenDependencyModuleArtifact(it)
+        }
+        val sourceJars = lib.sources.map { uri -> uri.toString() }.map {
+            val artifact = MavenDependencyModuleArtifact(it)
+            artifact.classifier = "sources"
+            artifact
+        }
+
+        // Matches the Maven group (organization), artifact, and version in the Bazel dependency
+        // string such as .../execroot/monorepo/bazel-out/k8-fastbuild/bin/external/maven/com/google/guava/guava/31.1-jre/processed_guava-31.1-jre.jar
+        val regexPattern = """.*/maven/(.+)/([^/]+)/([^/]+)/[^/]+.jar""".toRegex()
```

Comment:
As far as I know there is no other way to get the coordinates, but the current regex seems to work and also the actual data is not used by Metals at all, just the relationship between source and normal jars.

Tested it on a large repo and it seems to work fine.

Reply:
okay, I'm 99% sure it doesn't work with bzlmod.

Paths with bzlmod enabled will look something like `bazel-out/k8-fastbuild/bin/external/rules_jvm_external~override~maven~maven/commons-cli/commons-cli/1.6.0/commons-cli-1.6.0.jar`

Reply:
So in case of bzlmod different versions of rules_jvm_external will use different paths, + changing the name of the repo generated by `maven.install` will also affect the path.

The only stable thing is `~maven~` part, which contains the name of the extension.

See `bazel-out/k8-fastbuild/bin/external/rules_jvm_external~~maven~name/v1/https/repo1.maven.org/maven2/com/google/auto/service/auto-service-annotations/1.1.1/header_auto-service-annotations-1.1.1.jar` for example

Reply:
I changed the logic to use label to get the start of the organisation name and then the regex seems to work in both cases. Not perfect by any means, but I don't know if we can actually get organisation anyway else.